### PR TITLE
Refactor computer test 

### PR DESCRIPTION
**Pull request by fabientownsend**
Refactorisation of the computer class and its tests.

During this refactorisation I deleted the UI dependency of the computer to respec the Single Responsability.

I extract the algorithm from Computer into a new class MoveEvaluator. I did this in order to respect two principles:
- Single Responsibility
- Open Close

During this refactoring I exposed the returned value from move_evaluation, I tested that before but directly on my private Alpha Beta method. This way I respect the encapsulation of my algorithm and keep my tests.

### CODE REVIEW

**Comment by christophgockel** [comment]
I don't really understand how coveralls works. There are failures but it still runs coverage?

**Comment by fabientownsend** [comment]
I'm not really sure, I think it is based on the branch and not master

File: lib/move_evaluator.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  end
+
+  def maximing_player(current_mark, alpha, beta)
+    maximum_value = -100
+
+    board.free_positions.each do |position|
+      new_value = get_value_move(current_mark, alpha, beta, position)
+
+      maximum_value = max(maximum_value, new_value)
+      alpha = max(alpha, maximum_value)
```
Comment (christophgockel) [inline lib/move_evaluator.rb]: no biggie, but this could be `[alpha, maximum_value].max` if you want to save one private method.

Comment (fabientownsend) [reply lib/move_evaluator.rb]: I did the change, it would be weird to let it like that

**Pull request merged**